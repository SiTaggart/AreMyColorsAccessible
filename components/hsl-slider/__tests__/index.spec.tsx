/* eslint-env jest */

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, ShallowWrapper, ReactWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import HslSlider from '..';

describe('hsl-slider', () => {
  let onChangeMock: jest.Mock;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    onChangeMock = jest.fn();
  });

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<HslSlider id="input-id" onChange={onChangeMock} value="#ccc" />);
  });

  it('renders without crashing', () => {
    ReactDOM.render(
      <HslSlider id="input-id" onChange={onChangeMock} value="#ccc" />,
      document.createElement('div')
    );
    const hslSliderComp = renderer
      .create(<HslSlider id="input-id" onChange={onChangeMock} value="#ccc" />)
      .toJSON();
    expect(hslSliderComp).toMatchSnapshot();
  });

  it('should render a compact vaiant ', () => {
    const hslSliderComp = renderer
      .create(<HslSlider id="input-id" onChange={onChangeMock} value="#ccc" variant="compact" />)
      .toJSON();
    expect(hslSliderComp).toMatchSnapshot();
  });

  it('should call onchange callback when hue changed', () => {
    wrapper.find('#input-id-Hue').simulate('input', { target: { value: '60' } });
    expect(onChangeMock).toBeCalledWith('#CCCCCC', 'input-id');
  });

  it('should call onchange callback when saturation changed', () => {
    wrapper.find('#input-id-Saturation').simulate('input', { target: { value: '60' } });
    expect(onChangeMock).toBeCalledWith('#EBADAD', 'input-id');
  });

  it('should call onchange callback when lightness changed', () => {
    wrapper.find('#input-id-Lightness').simulate('input', { target: { value: '60' } });
    expect(onChangeMock).toBeCalledWith('#999999', 'input-id');
  });
});
