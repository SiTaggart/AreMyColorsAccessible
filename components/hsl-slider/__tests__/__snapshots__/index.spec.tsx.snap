// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hsl-slider renders without crashing 1`] = `
<div
  className="form-hsl-sliders"
>
  <div
    className="form-hsl-slider"
  >
    <label
      className="form-label"
      htmlFor="input-id-Hue"
    >
      Hue 0Â°
    </label>
    <input
      className="form-range"
      id="input-id-Hue"
      max={360}
      min={0}
      onChange={[Function]}
      onInput={[Function]}
      type="range"
      value={0}
    />
  </div>
  <div
    className="form-hsl-slider"
  >
    <label
      className="form-label"
      htmlFor="input-id-Saturation"
    >
      Saturation 0%
    </label>
    <input
      className="form-range"
      id="input-id-Saturation"
      max={100}
      min={0}
      onChange={[Function]}
      onInput={[Function]}
      type="range"
      value={0}
    />
  </div>
  <div
    className="form-hsl-slider"
  >
    <label
      className="form-label"
      htmlFor="input-id-Lightness"
    >
      Lightness 80%
    </label>
    <input
      className="form-range"
      id="input-id-Lightness"
      max={100}
      min={0}
      onChange={[Function]}
      onInput={[Function]}
      type="range"
      value={80}
    />
  </div>
</div>
`;

exports[`hsl-slider should render a compact vaiant  1`] = `
<div
  className="form-hsl-sliders form-hsl-sliders--compact"
>
  <div
    className="form-hsl-slider form-hsl-slider--compact"
  >
    <label
      className="form-label"
      htmlFor="input-id-Hue"
    >
      H
    </label>
    <input
      className="form-range"
      id="input-id-Hue"
      max={360}
      min={0}
      onChange={[Function]}
      onInput={[Function]}
      type="range"
      value={0}
    />
  </div>
  <div
    className="form-hsl-slider form-hsl-slider--compact"
  >
    <label
      className="form-label"
      htmlFor="input-id-Saturation"
    >
      S
    </label>
    <input
      className="form-range"
      id="input-id-Saturation"
      max={100}
      min={0}
      onChange={[Function]}
      onInput={[Function]}
      type="range"
      value={0}
    />
  </div>
  <div
    className="form-hsl-slider form-hsl-slider--compact"
  >
    <label
      className="form-label"
      htmlFor="input-id-Lightness"
    >
      L
    </label>
    <input
      className="form-range"
      id="input-id-Lightness"
      max={100}
      min={0}
      onChange={[Function]}
      onInput={[Function]}
      type="range"
      value={80}
    />
  </div>
</div>
`;
